demo/full/scripts/controllers/ContentList.tsx

```diff
@@ -419,7 +400,7 @@ function ContentList({
   }, []);
 
   React.useEffect(() => {
-    const parsedHash = parseHashInURL(location.hash);
+    const parsedHash = parseHashInURL(decodeURI(location.hash));
```

Comment:
I don't think URL fragments have to be URL-encoded, do they?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Location/hash

> The fragment is not [URL decoded](https://en.wikipedia.org/wiki/URL_encoding)

It is encoded :), but it was not noticeable until now as the previous hash doesn't have any character that needs to be encoded. But since we have now object of form `key={"objectKey": "objectValue"}` in the URL,  some characters needs to be encoded such as `"` that is encoded as `%22`

Reply:
> The fragment is not [URL decoded](https://en.wikipedia.org/wiki/URL_encoding)

I'm under the impression that your documentation does not infirm my statement. And some random answers on the internet (https://stackoverflow.com/questions/77375436/must-the-percent-character-be-percent-encoded-in-a-url-fragment) go my way.

RFCs are sadly difficult to follow here because there's too many for URIs.

A more interesting answer from https://stackoverflow.com/questions/20679500/how-to-encode-space-in-the-fragment-identifier-in-a-url indicate that it depends but for HTML page it should be percent-encoded. However his justification doesn't apply here because he relies on HTML5 "Fragment navigations" algorithm which do not apply here (we're not trying to navigate to an HTML node).

All in all it seems to me that it's hard to understand if percent-encoding should be considered mandatory or not here. In doubt, I'm OK with doing it.

Reply:
I think the browser is doing things on it's own and encode the hash when setting location.hash:

```js
location.hash = '{"foo":"bar"}';
console.log(location.hash) // '#{%22foo%22:%22bar%22}'
JSON.parse(location.hash) // Error location.hash is not a valid JSON
JSON.parse(decodeURI(location.hash)) // OK
```

Reply:
:O Ok interesting 

I dived into the HTML spec and it actually tells us this.

1. For the `hash` setter https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-location-hash :

> Let input be the given value with a single leading "#" removed, if any.
> Set copyURL's [fragment](https://url.spec.whatwg.org/#concept-url-fragment) to the empty string.
> [Basic URL parse](https://url.spec.whatwg.org/#concept-basic-url-parser) input, with copyURL as [url](https://url.spec.whatwg.org/#basic-url-parser-url) and [fragment state](https://url.spec.whatwg.org/#fragment-state) as [state override](https://url.spec.whatwg.org/#basic-url-parser-state-override).

So it seems that the `Basic URL parse` algorithm applies to the URL + what you set to `location.hash` (interestingly removing if present a leading `#`).
The `Basic URL parse` algorithm itself is too convoluted to me for just checking what it does but it is highly probable that for the fragment it goes here in its state machine:  https://url.spec.whatwg.org/#fragment-state

It seems that when encountering a fragment, the browser should:

> [UTF-8 percent-encode](https://url.spec.whatwg.org/#utf-8-percent-encode) [c](https://url.spec.whatwg.org/#c) using the [fragment percent-encode set](https://url.spec.whatwg.org/#fragment-percent-encode-set) and append the result to url’s [fragment](https://url.spec.whatwg.org/#concept-url-fragment).

That's where it goes to be the most interesting, the percent encoding for fragments is actually specific to fragments and is defined here:
https://url.spec.whatwg.org/#fragment-percent-encode-set

It inherits C0-control-percent-encode-set: https://url.spec.whatwg.org/#c0-control-percent-encode-set

Which combines:
  1. A C0 control is a [code point](https://infra.spec.whatwg.org/#code-point) in the range U+0000 NULL to U+001F INFORMATION SEPARATOR ONE, inclusive.
  2. All code points superior to U+007E (~).

This makes me want to over-engineer that thing to specifically percent-decode fragments ^^
Does that seem fun to you? If not, it does to me so I could do it also.

It surely would really be never really useful but sometimes developers just want to flex.

Reply:
Interestingly, `%` is not percent encoded. So it's possible to lose information after percent encoding and decoding a string:

```js
location.hash = "toto %20 titi";
console.log("toto %20 titi"); // => "toto%20%20%20titi" => decode => "toto   titi"
```

HTML is broken to the core

Reply:
I think the simpler is to encode it before setting the hash:

```js 
location.hash = encodeURI("toto %20 titi");
console.log(location.hash); // => "toto%20%2520%20titi"
console.log(decodeURI(location.hash)); // => "toto %20 titi'"

```

Reply:
Yeah it is simpler :p  

And it even percent encode a percent as expected. Though it's less 'fun' and it makes the URL less human-readable but it is still functional so OK

Apparently `encodeURI` is from JavaScript this time (ECMA262) and it encodes everything that's not:
> ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_-.!~*'();/?:@&=+$,#